src/core/metrics.cc

```diff
@@ -172,20 +172,42 @@ Metrics::InitializeNvmlMetrics()
     return false;
   }
 
-  unsigned int dcnt;
-  nvmlerr = nvmlDeviceGetCount(&dcnt);
-  if (nvmlerr != NVML_SUCCESS) {
-    LOG_ERROR << "failed to get device count for nvml metrics: NVML_ERROR "
-              << nvmlerr;
+  int dcnt;
+  cudaError_t cudaerr = cudaGetDeviceCount(&dcnt);
+  if (cudaerr != cudaSuccess) {
+    LOG_ERROR << "failed to get device count for cuda devices: CUDA_ERROR "
+              << cudaerr;
     return false;
   }
 
   // Create NVML metrics for each GPU
-  LOG_INFO << "found " << dcnt << " GPUs supporting NVML metrics";
-  for (unsigned int didx = 0; didx < dcnt; ++didx) {
+  LOG_INFO << "found " << dcnt << " CUDA visible devices.";
+  for (int didx = 0; didx < dcnt; ++didx) {
     // Get handle for the GPU
+    cudaDeviceProp gpu_properties;
+    cudaError_t cudaerr = cudaGetDeviceProperties(&gpu_properties, didx);
+    if (cudaerr != cudaSuccess) {
+      LOG_ERROR << "failed to get device properties for cuda device " << didx << "CUDA_ERROR "
+                << cudaerr;
+      return false;
```

Comment:
These 3 "return false" should be continue so that we just skip GPUs that we have trouble with. But then dcnt gets out of sync with the number of entries in gpu_utilization_, etc. What we should instead do is at live 236 save the "nvmlDevice_t gpu" value into nvml_device_ to that it is parallel to gpu_utilization_, etc. Then at line 246 and later we don't use dcnt but instead use the nvml_device_ and other vectors that all have the same length.